src/main/java/se/sundsvall/billingpreprocessor/service/creator/ExternalInvoiceCreator.java

```diff
@@ -0,0 +1,85 @@
+package se.sundsvall.billingpreprocessor.service.creator;
+
+import static java.util.Optional.ofNullable;
+import static org.springframework.util.CollectionUtils.isEmpty;
+import static se.sundsvall.billingpreprocessor.Constants.EMPTY_ARRAY;
+import static se.sundsvall.billingpreprocessor.Constants.EXTERNAL_INVOICE_TYPE;
+import static se.sundsvall.billingpreprocessor.Constants.GENERATING_SYSTEM;
+import static se.sundsvall.billingpreprocessor.service.creator.config.InvoiceCreatorConfig.EXTERNAL_INVOICE_BUILDER;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toCustomer;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toFileHeader;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toInvoiceAccountingRow;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toInvoiceDescriptionRows;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toInvoiceFooter;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toInvoiceHeader;
+import static se.sundsvall.billingpreprocessor.service.mapper.ExternalInvoiceMapper.toInvoiceRow;
+import static se.sundsvall.billingpreprocessor.service.util.ProblemUtil.createProblem;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.List;
+
+import org.beanio.BeanWriter;
+import org.beanio.StreamFactory;
+import org.beanio.builder.StreamBuilder;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import se.sundsvall.billingpreprocessor.integration.db.model.BillingRecordEntity;
+import se.sundsvall.billingpreprocessor.integration.db.model.InvoiceRowEntity;
+import se.sundsvall.billingpreprocessor.service.util.StringUtil;
+
+@Component
+public class ExternalInvoiceCreator {
```

Comment:
Method names in ExternalVoiceCreator and InternalVoiceCreator is very vague.

"toBytes" - indicates some kind of conversion to a byte array.
"handleInvoice" - could be anything really.

Reply:
- ExternalInvoiceCreator: is what it is - a creator for external (or internal for the InternalInvoiceCreator) invoices. I do not know a better name for it, any suggestions?
- toBytes: returns a byte-array (which the metod indicates). It's following the same method name nomenclature as we have in mappers - to<type>(class input). Care to elaborate on an alternative method name?
- handleInvoice: I can rename it to something like "processInvoice", but I cant see that it will do much. 

I'll extend creator-classes with more documentation to make it more clear of what their purpose is.